lumicks/pylake/population/mixture.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+from sklearn.mixture import GaussianMixture
+from scipy import stats
+
+
+def as_sorted(fcn):
+    """Decorator to return results sorted according to mapping array."""
+
+    def wrapper(self, *args, **kwargs):
+        result = fcn(self, *args, **kwargs)
+        return result[self._map]
+
+    return wrapper
+
+
+class GaussianMixtureModel:
+    def __init__(self, data, K, init_method, n_init, tol, max_iter):
+        """A wrapper around scikit-learn's GMM
+
+        Parameters
+        ----------
+        K : int
+            number of Gaussian components in the model
+        """
+        self.K = K
+        self._model = GaussianMixture(
+            n_components=K, init_params=init_method, n_init=n_init, tol=1e-3, max_iter=100
+        )
+        self._model.fit(data)
+
+    @classmethod
+    def from_slice(cls, slc, K, init_method="kmeans", n_init=1, tol=1e-3, max_iter=100):
+        data = slc.data.reshape((-1, 1))
+        return cls(data, K, init_method="kmeans", n_init=1, tol=1e-3, max_iter=100)
+
+    @property
+    def exit_flag(self):
+        return {
+            "converged": self._model.converged_,
+            "n_iter": self._model.n_iter_,
+            "lower_bound": self._model.lower_bound_,
+        }
```

Comment:
I find finding out about convergence after the fact a bit weird. When the model fails to converge, shouldn't fit throw?

What value do you get for `converged_` if you did multiple starts?

This property also seems undocumented and untested.

Reply:
I think this is pretty standard behavior for fitting (the values are actually pulled directly from the sklearn GMM instance after it is fit). I think throwing an exception is a bit too strict. This way the results are available to the user and they can decide what to do with it (throw if `converged == False` for example)